packages/in_app_purchase/example/android/app/src/test/java/io/flutter/plugins/inapppurchase/TranslatorTest.java

```diff
@@ -87,33 +118,44 @@ public void fromPurchasesList_null() {
   public void fromPurchasesResult() throws JSONException {
     PurchasesResult result = mock(PurchasesResult.class);
     final String purchase2Json =
-        "{\"orderId\":\"foo2\",\"packageName\":\"bar\",\"productId\":\"consumable\",\"purchaseTime\":11111111,\"purchaseState\":0,\"purchaseToken\":\"baz\"}";
+        "{\"orderId\":\"foo2\",\"packageName\":\"bar\",\"productId\":\"consumable\",\"purchaseTime\":11111111,\"purchaseState\":0,\"purchaseToken\":\"baz\",\"developerPayload\":\"dummy payload\",\"isAcknowledged\":\"true\"}";
     final String signature = "signature";
     final List<Purchase> expectedPurchases =
         Arrays.asList(
             new Purchase(PURCHASE_EXAMPLE_JSON, signature), new Purchase(purchase2Json, signature));
     when(result.getPurchasesList()).thenReturn(expectedPurchases);
-    when(result.getResponseCode()).thenReturn(BillingResponse.OK);
-
+    when(result.getResponseCode()).thenReturn(BillingClient.BillingResponseCode.OK);
+    BillingResult newBillingResult =
+        BillingResult.newBuilder()
+            .setDebugMessage("dummy debug message")
+            .setResponseCode(BillingClient.BillingResponseCode.OK)
+            .build();
+    when(result.getBillingResult()).thenReturn(newBillingResult);
     final HashMap<String, Object> serialized = Translator.fromPurchasesResult(result);
 
-    assertEquals(BillingResponse.OK, serialized.get("responseCode"));
+    assertEquals(BillingClient.BillingResponseCode.OK, serialized.get("responseCode"));
     List<Map<String, Object>> serializedPurchases =
         (List<Map<String, Object>>) serialized.get("purchasesList");
     assertEquals(expectedPurchases.size(), serializedPurchases.size());
     assertSerialized(expectedPurchases.get(0), serializedPurchases.get(0));
     assertSerialized(expectedPurchases.get(1), serializedPurchases.get(1));
+
+    Map<String, Object> billingResultMap = (Map<String, Object>) serialized.get("billingResult");
+    assertEquals(billingResultMap.get("responseCode"), newBillingResult.getResponseCode());
+    assertEquals(billingResultMap.get("debugMessage"), newBillingResult.getDebugMessage());
   }
 
   @Test
-  public void fromPurchasesResult_null() throws JSONException {
-    PurchasesResult result = mock(PurchasesResult.class);
-    when(result.getResponseCode()).thenReturn(BillingResponse.ERROR);
-
-    final HashMap<String, Object> serialized = Translator.fromPurchasesResult(result);
-
-    assertEquals(BillingResponse.ERROR, serialized.get("responseCode"));
-    assertEquals(Collections.emptyList(), serialized.get("purchasesList"));
+  public void fromBillingResult() throws JSONException {
+    BillingResult newBillingResult =
+        BillingResult.newBuilder()
+            .setDebugMessage("dummy debug message")
```

Comment:
nit: Can this ever be null? Would it be worth adding a test for this to make sure the translator can handle serializing it still?